src/ImageProcessor.Web/HttpModules/ImageProcessingModule.cs

```diff
@@ -826,7 +826,10 @@ private IImageService GetImageServiceForRequest(string url, string applicationPa
             foreach (IImageService service in services)
             {
                 string key = service.Prefix;
-                if (!string.IsNullOrWhiteSpace(key) && path.StartsWith(key, StringComparison.InvariantCultureIgnoreCase))
+                if (!string.IsNullOrWhiteSpace(key) 
+                    && (service.IsRegex 
```

Comment:
How about a logical or here instead? That way you could prevent altering the interface.

```
&& (Regex.IsMatch(path, key, RegexOptions.IgnoreCase) 
        || path.StartsWith(key, StringComparison.InvariantCultureIgnoreCase)
```